### PR TITLE
Add cache for Nuget packages

### PR DESCRIPTION
**Pull request by AlexNDRmac**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [X] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [X] I have provided test coverage for my change (where applicable)

### Description

- Added cache action for Nuget packages for speedup Nuget restore before each CI builds

### CODE REVIEW

**Comment by AlexNDRmac** [comment]
@antonfirsov current summary restoring time for all projects is about ~1 minutes. With cache (next time, when you will build with enabled cache action) summary restore time should be ~5-10 seconds.
I guess, you can compare build time for each OS by triggering build action again.
Cache key based on all files in repository with .csproj extension, so any changes with packages will be change hash of cache kay.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -56,6 +56,14 @@ jobs:
                   git fetch --prune --unshallow
                   git submodule -q update --init --recursive
 
+            - name: Setup nuget cache
+              uses: actions/cache@v2
+              id: nuget-cache
+              with:
+                  path: ~/.nuget
+                  key: ${{ runner.os }}-nuget-${{ hashFiles('**/ImageSharp*.csproj') }}
```
Comment (tocsoft) [inline .github/workflows/build-and-test.yml:64]: you will need to add the `*.props`/`*.targets` too as we define most of our package versions in those shared locations i.e. https://github.com/SixLabors/ImageSharp/blob/0e0dc2ae9cafcdf5bde9d185919cf073ddf4f186/Directory.Build.targets#L19-L37

**Review by tocsoft — CHANGES_REQUESTED** [review]
haven't captured all the files that could end up changing the packages

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (AlexNDRmac) [reply .github/workflows/build-and-test.yml]: Thanks. Added with next commit.

**Review by JimBobSquarePants — APPROVED** [review]
Ok. Looks like everything is covered here now. Let's give it a whirl!

**Pull request merged**